st3/lsp_utils/npm_client_handler.py

```diff
@@ -44,6 +59,26 @@ def send_request(self, method: str, params: Any, handler: Callable[[Optional[str
         self.__client.send_request(
             Request(method, params), lambda result: handler(result, False), lambda result: handler(result, True))
 
+    @staticmethod
+    def as_notification_handler(event_name: str) -> Callable[[_T], _T]:
+        """ Marks the decorated function as a handler for the notification event. """
+
+        def decorator(func: _T) -> _T:
+            setattr(func, ApiWrapper.HANDLER_MARK_NOTIFICATION, event_name)
```

Comment:
Should put the registered event_name into a list, otherwise if this decorator is used twice on a same function, only the later one works. But, well, just sharing an idea here :D